### PR TITLE
Task 6 - Map class with getIsland() method

### LINKED ISSUE #8: Task 6 - Map (function getIsland)
W klasie Map napisz funkcję

Island* getIsland(const Island::Coordinates& coordinate)

Powinna ona przeszukać std::vector<Island> i zwrócić szukaną wyspę.

### PR DESCRIPTION
**Pull request by mariuszlisowski**

### CODE REVIEW

File: shm/map.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#include "map.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <random>
+
+constexpr size_t MIN_COORDINATE = 0; // already lowest
```
Comment (Petrus9107) [inline shm/map.cpp]: would consider possibility of using negative coordinates (up for discussion)

File: shm/map.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        if (it == vectorOfIslands_.end()) {
+            vectorOfIslands_.push_back(Island(posX, posY));
+        }
+    }
+}
+
+Map::~Map() {}
+
+Island* Map::getIsland(const Coordinates& coordinates) {
+    auto itr = std::find_if(vectorOfIslands_.begin(),
```
Comment (Petrus9107) [inline shm/map.cpp]: lets try to be consistent and use it or itr

File: shm/map.hpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+#include "island.hpp"
+
+#include <vector>
+
+class Map {
+public:
+    Map();
+    virtual ~Map();
+    Island *getIsland(const Coordinates&);
```
Comment (Petrus9107) [inline shm/map.hpp]: I guess it's better to include the argument name in method signature as well

**Review by Petrus9107 — COMMENTED** [review]
Anyways, Good Job!

File: shm/map.cpp (hunk 1 above)
Comment (mariuszlisowski) [reply shm/map.cpp]: Let the coordinate system start with (0, 0) :)

File: shm/map.cpp (hunk 2 above)
Comment (mariuszlisowski) [reply shm/map.cpp]: Good point. Done.

File: shm/map.hpp (hunk 3 above)
Comment (mariuszlisowski) [reply shm/map.hpp]: I don't know what the good practice is. Adding the argument name makes it more readable but omitting one makes it shorter. Your suggestion wins anyway.

**Pull request merged**